pkg/eventlog/fs/pointer_registry.go

```diff
@@ -0,0 +1,80 @@
+package fs
+
+// NOTE: for encoding and decoding data from the file
+// encoding/json has to be used over encoding/gob due to a possible bug
+// in afero. see issue #172 for reference
+// https://github.com/spf13/afero/issues/172
+import (
+	"encoding/json"
+	"io"
+	"os"
+
+	"github.com/gomods/athens/pkg/eventlog"
+	"github.com/spf13/afero"
+)
+
+// Registry is a pointer registry for olympus server event logs
+type Registry struct {
+	rootDir string
+	fs      afero.Fs
+}
+
+// registryData is a map[string]string used to encode/decode the registry from disk
+type registryData map[string]string
+
+var registryFilename = "pointerRegistry"
+
+// NewRegistry returns a file based implementation of a pointer registry
+func NewRegistry(rootDir string, filesystem afero.Fs) *Registry {
+	return &Registry{rootDir: rootDir, fs: filesystem}
+}
+
+// LookupPointer returns the pointer to the given deployment's event log
+func (r *Registry) LookupPointer(deploymentID string) (string, error) {
+	f, err := r.fs.OpenFile(registryFilename, os.O_RDONLY|os.O_CREATE, 0440)
+	if err != nil {
+		return "", err
+	}
+
+	var data = make(registryData)
+
+	dec := json.NewDecoder(f)
+	if err := dec.Decode(&data); err != nil {
+		return "", err
+	}
+
+	result, ok := data[deploymentID]
+	if !ok {
+		return "", eventlog.ErrDeploymentNotFound
+	}
+
+	return result, f.Close()
```

Comment:
Question: wouldn't the file stay open forever if we get an error doing decoding or ErrDeploymentNotFound? The same inside the SetPointer func

Reply:
yes indeed good catch. @robjloranger can you put defer close right after error check for Open?